pkg/infradb/utils.go

```diff
@@ -0,0 +1,294 @@
+				}
+
+				// tempSubs holds the subscribers list to be contacted for every VRF object each time
+				// for replay
+				tempSubs := lb.prepareObjectsForReplay(componentName, lbSubs)
+
+				err = infradb.client.Set(lb.Name, lb)
+				if err != nil {
+					return nil, nil, err
+				}
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, lb)
+			}
+		case "svi":
+			svisMap := make(map[string]bool)
+			found, err := infradb.client.Get("svis", &svisMap)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			if !found {
+				log.Println("gatherObjectsAndSubsToReplay(): No SVIs have been found")
+				continue
+			}
+			for key := range svisMap {
+				svi := &Svi{}
+				found, err := infradb.client.Get(key, svi)
+				if err != nil {
+					return nil, nil, err
+				}
+
+				if !found {
+					return nil, nil, ErrKeyNotFound
+				}
+
+				// tempSubs holds the subscribers list to be contacted for every VRF object each time
+				// for replay
+				tempSubs := svi.prepareObjectsForReplay(componentName, sviSubs)
+
+				err = infradb.client.Set(svi.Name, svi)
+				if err != nil {
+					return nil, nil, err
+				}
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, svi)
+			}
+		case "bp":
+			bpsMap := make(map[string]bool)
+			found, err := infradb.client.Get("bps", &bpsMap)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			if !found {
+				log.Println("gatherObjectsAndSubsToReplay(): No Bridge Ports have been found")
+				continue
+			}
+			for key := range bpsMap {
+				bp := &BridgePort{}
+				found, err := infradb.client.Get(key, bp)
+				if err != nil {
+					return nil, nil, err
+				}
+
+				if !found {
+					return nil, nil, ErrKeyNotFound
+				}
+
+				// tempSubs holds the subscribers list to be contacted for every VRF object each time
+				// for replay
+				tempSubs := bp.prepareObjectsForReplay(componentName, bpSubs)
+
+				err = infradb.client.Set(bp.Name, bp)
+				if err != nil {
+					return nil, nil, err
+				}
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, bp)
+			}
+		}
+	}
+
+	return objectsToReplay, subsForReplay, nil
+}
+
+// createReplayTasks create new tasks for the realization of the new replay objects intents
+func createReplayTasks(objectsToReplay []interface{}, subsForReplay [][]*eventbus.Subscriber) {
+	for i, obj := range objectsToReplay {
+		switch tempObj := obj.(type) {
+		case *Vrf:
+			taskmanager.TaskMan.CreateTask(tempObj.Name, "vrf", tempObj.ResourceVersion, subsForReplay[i])
+		case *LogicalBridge:
+			taskmanager.TaskMan.CreateTask(tempObj.Name, "logical-bridge", tempObj.ResourceVersion, subsForReplay[i])
+		case *Svi:
+			taskmanager.TaskMan.CreateTask(tempObj.Name, "svi", tempObj.ResourceVersion, subsForReplay[i])
+		case *BridgePort:
+			taskmanager.TaskMan.CreateTask(tempObj.Name, "bridge-port", tempObj.ResourceVersion, subsForReplay[i])
+		}
```

Comment:
what if we receive type not listed in this switch?

Reply:
The types of objects are just 4. Always. I do not see any case where we will receive any other type.

Reply:
we can receive if we have a mistake -> why not handle such a case instead of crashing

Reply:
I can put a "default" case if that is necessary but I can not see how we will receive any other type except from those 4 types that we allready have.

Reply:
Done